linkerd/app/outbound/src/http/logical/policy/route/filters.rs

```diff
@@ -99,13 +114,32 @@ pub fn apply_http<B>(
             http::Filter::InternalError(msg) => {
                 return Err(errors::HttpInvalidPolicy(msg).into());
             }
+            http::Filter::ResponseHeaders(_) => {} // ResponseHeaders filter does not apply to requests.
+        }
+    }
+
+    Ok(())
+}
+
+pub fn apply_http_response<B>(
+    filters: &[http::Filter],
+    rsp: &mut ::http::Response<B>,
+) -> Result<()> {
+    // TODO Do any metrics apply here?
+    for filter in filters {
+        match filter {
+            http::Filter::InjectFailure(_) => {} // InjectFailure filter does not apply to responses.
+            http::Filter::Redirect(_) => {}      // Redirect filter does not apply to responses.
+            http::Filter::RequestHeaders(_) => {} // RequestHeaders filter does not apply to responses.
+            http::Filter::InternalError(_) => {} // InternalError filter does not apply to responses.
```

Comment:
this feels...somewhat unfortunate. it would be preferable if we could iterate over all the filters once, and apply them not just to the request but to the *future*, so that the filter can act on both the request and the response as it sees fit. but, that would require a much more substantial redesign of how we represent filters, essentially turning them into arbitrary middleware, and then we run into issues with the future type being determined from the value of a dynamic list of filters...